version: "3.8"

services:
  web:
    build:
      context: ./
      dockerfile: ./deploy/docker/app/Dockerfile
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - ./:/usr/src/app/
    ports:
      - 8000:8000
    env_file:
      - ./deploy/docker/.env.dev
    depends_on:
      db:
        condition: service_healthy

    # links:
    #   - db
    # networks:
    #   - django-dev-net
  db:
    image: postgres:12.0-alpine
    # networks:
    #   - django-dev-net
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_USER=immersion
      - POSTGRES_PASSWORD=immersion
      - POSTGRES_DB=immersion
    # ports:
    #   - "5432:5432"
    healthcheck:
      test: "pg_isready --username=immersion && psql --username=immersion --list"
      timeout: 30s
      retries: 20

volumes:
  postgres_data:
# networks:
#   django-dev-net:
# driver: bridge
